pgjdbc/src/main/java/org/postgresql/jdbc/PgCallableStatement.java

```diff
@@ -691,7 +696,143 @@ public Reader getCharacterStream(String parameterName) throws SQLException {
+      }
+    } else if (type == Long.class) {
+      if (sqlType == Types.BIGINT) {
+        long longValue = getLong(parameterIndex);
+        if (wasNull()) {
+          return null;
+        }
+        return type.cast(longValue);
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Float.class) {
+      if (sqlType == Types.REAL) {
+        float floatValue = getFloat(parameterIndex);
+        if (wasNull()) {
+          return null;
+        }
+        return type.cast(floatValue);
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Double.class) {
+      if (sqlType == Types.FLOAT || sqlType == Types.DOUBLE) {
+        double doubleValue = getDouble(parameterIndex);
+        if (wasNull()) {
+          return null;
+        }
+        return type.cast(doubleValue);
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Date.class) {
+      if (sqlType == Types.DATE) {
+        return type.cast(getDate(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Time.class) {
+      if (sqlType == Types.TIME) {
+        return type.cast(getTime(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Timestamp.class) {
+      if (sqlType == Types.TIMESTAMP
+              //#if mvn.project.property.postgresql.jdbc.spec >= "JDBC4.2"
+              || sqlType == Types.TIMESTAMP_WITH_TIMEZONE
+      //#endif
+      ) {
+        return type.cast(getTimestamp(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == Array.class) {
+      if (sqlType == Types.ARRAY) {
+        return type.cast(getArray(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == SQLXML.class) {
+      if (sqlType == Types.SQLXML) {
+        return type.cast(getSQLXML(parameterIndex));
+      } else {
+        throw new PSQLException(GT.tr("Unsupported type conversion to {1}.", type),
+                PSQLState.INVALID_PARAMETER_VALUE);
+      }
+    } else if (type == UUID.class) {
+      return type.cast(getObject(parameterIndex));
+    } else if (type == InetAddress.class) {
```

Comment:
This is not in the specification, but it's fine as an additional extension support.

Reply:
I'll back out of this because of #1567